core/src/main/java/org/elasticsearch/plugins/PluginInfo.java

```diff
@@ -95,6 +95,10 @@ public static PluginInfo readFromProperties(Path dir) throws IOException {
         if (version == null) {
             throw new IllegalArgumentException("Property [version] is missing for plugin [" + name + "]");
         }
+        String pluginName = props.getProperty("plugin.name");
+        if (pluginName == null) {
+            throw new IllegalArgumentException("Property [plugin.name] is missing for plugin [" + name + "]");
+        }
```

Comment:
@rmuir added element to the enforcer plugin, but I'm not sure what check to insert in pluginservice and how that service actually works. Can this be added in a follow up PR?